### ISSUE TITLE
[Ray Client] Can't connect to Ray client locally

### ISSUE THREAD

**Issue by architkulkarni** [ray-project/ray#14393]
### What is the problem?
Connecting to the Ray client after `ray start --head`doesn't seem to work, see the repro for details.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Master, Python 3.6, Mac OS
### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

First call `ray start --head`, then run

```python
import ray, ray.util
ray.util.connect("localhost:10001")
```

This yields the error
```python-traceback
The autoscaler failed with the following error:
Traceback (most recent call last):
  File "/Users/archit/ray/python/ray/monitor.py", line 272, in run
    self._run()
  File "/Users/archit/ray/python/ray/monitor.py", line 166, in _run
    self.update_load_metrics()
  File "/Users/archit/ray/python/ray/monitor.py", line 145, in update_load_metrics
    ip = resource_message.node_manager_address
AttributeError: node_manager_address

Got Error from data channel -- shutting down: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Exception iterating responses: Protocol message ConnectionInfoResponse has no "protocol_version" field."
        debug_error_string = "{"created":"@1614364099.647061000","description":"Error received from peer ipv6:[::1]:10001","file":"src/core/lib/surface/call.cc","file_line":1055,"grpc_message":"Exception iterating responses: Protocol message ConnectionInfoResponse has no "protocol_version" field.","grpc_status":2}"
>
Exception in thread Thread-6:
Traceback (most recent call last):
  File "/Users/archit/anaconda3/envs/ray-py36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/archit/anaconda3/envs/ray-py36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/archit/ray/python/ray/util/client/dataclient.py", line 87, in _data_main
    raise e
  File "/Users/archit/ray/python/ray/util/client/dataclient.py", line 62, in _data_main
    for response in resp_stream:
  File "/Users/archit/anaconda3/envs/ray-py36/lib/python3.6/site-packages/grpc/_channel.py", line 416, in __next__
    return self._next()
  File "/Users/archit/anaconda3/envs/ray-py36/lib/python3.6/site-packages/grpc/_channel.py", line 706, in _next
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Exception iterating responses: Protocol message ConnectionInfoResponse has no "protocol_version" field."
        debug_error_string = "{"created":"@1614364099.647061000","description":"Error received from peer ipv6:[::1]:10001","file":"src/core/lib/surface/call.cc","file_line":1055,"grpc_message":"Exception iterating responses: Protocol message ConnectionInfoResponse has no "protocol_version" field.","grpc_status":2}"
>

---------------------------------------------------------------------------
ConnectionError                           Traceback (most recent call last)
<ipython-input-2-cd3f97e783f7> in <module>
----> 1 ray.util.connect("localhost:10001")

~/ray/python/ray/util/client_connect.py in connect(conn_str, secure, metadata, connection_retries, ignore_version)
     30         metadata=metadata,
     31         connection_retries=3,
---> 32         ignore_version=ignore_version)
     33 
     34 

~/ray/python/ray/util/client/__init__.py in connect(self, conn_str, secure, metadata, connection_retries, ignore_version)
     68                 connection_retries=connection_retries)
     69             self.api.worker = self.client_worker
---> 70             conn_info = self.client_worker.connection_info()
     71             self._check_versions(conn_info, ignore_version)
     72             return conn_info

~/ray/python/ray/util/client/worker.py in connection_info(self)
    148     def connection_info(self):
    149         try:
--> 150             data = self.data_client.ConnectionInfo()
    151         except grpc.RpcError as e:
    152             raise e.details()

~/ray/python/ray/util/client/dataclient.py in ConnectionInfo(self, context)
    113         datareq = ray_client_pb2.DataRequest(
    114             connection_info=ray_client_pb2.ConnectionInfoRequest())
--> 115         resp = self._blocking_send(datareq)
    116         return resp.connection_info
    117 

~/ray/python/ray/util/client/dataclient.py in _blocking_send(self, req)
    104             if self._in_shutdown:
    105                 raise ConnectionError(
--> 106                     f"cannot send request {req}: data channel shutting down")
    107             data = self.ready_data[req_id]
    108             del self.ready_data[req_id]

ConnectionError: cannot send request req_id: 1
connection_info {
}
: data channel shutting down
```
If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by architkulkarni** [comment]
Here's the server log in /tmp:

```
INFO:ray.util.client.server.server:Starting Ray Client server on 0.0.0.0:10001
INFO:ray.util.client.server.dataservicer:New data connection from client 9c3f36a3132f48deb1c09c72cd5ab661
INFO:ray.util.client.server.logservicer:New logs connection
2021-03-01 09:22:17,882	INFO worker.py:665 -- Connecting to existing Ray cluster at address: 192.168.1.22:6379
2021-03-01 09:22:17,906	WARNING worker.py:1063 -- The autoscaler failed with the following error:
Traceback (most recent call last):
  File "/Users/archit/ray/python/ray/monitor.py", line 272, in run
    self._run()
  File "/Users/archit/ray/python/ray/monitor.py", line 166, in _run
    self.update_load_metrics()
  File "/Users/archit/ray/python/ray/monitor.py", line 145, in update_load_metrics
    ip = resource_message.node_manager_address
AttributeError: node_manager_address

2021-03-01 09:22:17,907	INFO dataservicer.py:70 -- Lost data connection from client 9c3f36a3132f48deb1c09c72cd5ab661
ERROR:grpc._server:Exception iterating responses: Protocol message ConnectionInfoResponse has no "protocol_version" field.
Traceback (most recent call last):
  File "/Users/archit/anaconda3/envs/ray-py36/lib/python3.6/site-packages/grpc/_server.py", line 453, in _take_response_from_response_iterator
    return next(response_iterator), True
  File "/Users/archit/ray/python/ray/util/client/server/dataservicer.py", line 61, in Datapath
    connection_info=self._build_connection_response())
  File "/Users/archit/ray/python/ray/util/client/server/dataservicer.py", line 89, in _build_connection_response
    protocol_version=CURRENT_PROTOCOL_VERSION)
ValueError: Protocol message ConnectionInfoResponse has no "protocol_version" field.

```

**Comment by simon-mo** [comment]
I can't repro it with ray master. what's your grpc version?
```
$ pip freeze | grep grpc     
grpcio==1.31.0
grpcio-opentracing==1.1.4
grpcio-tools==1.26.0
```

**Comment by architkulkarni** [comment]
```
❯ pip freeze | grep grpc
grpcio==1.31.0
```

**Comment by simon-mo** [comment]
hmm and ray commit? 
```
$ git log
commit 343ebf8ea7f0d0784fb9ccc629361aad8a375f8b (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Author: Raphael CHEN <raphaelchensi@outlook.com>
Date:   Tue Mar 2 00:14:39 2021 +0800

    [tune] Checkpoint according to nested metric (#14379)
```

**Comment by architkulkarni** [comment]
```
commit db0c16824c46bf7b7b098845d6ed9cfabca15815 (HEAD -> master, upstream/master, runtime-env-conda)
Author: Micah Yong <micahtyong@gmail.com>
Date:   Mon Mar 1 09:27:22 2021 -0800
```

**Comment by architkulkarni** [comment]
Hmm, let me try again with your commit

**Comment by architkulkarni** [comment]
My fault, I just had to rerun `bazel build //:ray_pkg`.
